controllers/chat/sendAnonymousMessage.js

```diff
@@ -65,6 +67,21 @@ const sendAnonymousMessage = async (req, res) => {
     if (createdChannel?.channel?.is_channel_blocked)
       return res.status(403).json(responseError('Channel is blocked'));
 
+    // Prevent if user try to chat with blocked user
+    let channelType = createdChannel?.channel?.channel_type;
+    if (channelType === 4 || channelType === 0) {
+      let signUser = await UsersFunction.findSignedUserId(User, req.userId);
+      const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(UserBlockedUser, signUser);
+      let better_channel_member = createdChannel?.channel?.better_channel_member;
+      const blockedIdsSet = new Set(blockedIds);
+      const isBlocked = better_channel_member.some(
+        (member) => member.user_id !== req.userId && blockedIdsSet.has(member.user_id)
+      );
+      if (isBlocked) {
+        return res.status(403).json(responseError('This user does not want to receive messages'));
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a user is blocked seems fine, but there are some performance considerations. You're fetching all blocked users and then checking if the recipient is in that list. This could be inefficient if the list of blocked users is large. Instead, you could directly query the database to check if a specific user is blocked. This would reduce the amount of data fetched from the database and improve performance.

Here's a rough idea of how you might do this:

```diff
- let signUser = await UsersFunction.findSignedUserId(User, req.userId);
- const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(UserBlockedUser, signUser);
- let better_channel_member = createdChannel?.channel?.better_channel_member;
- const blockedIdsSet = new Set(blockedIds);
- const isBlocked = better_channel_member.some(
-   (member) => member.user_id !== req.userId && blockedIdsSet.has(member.user_id)
- );
+ let recipientId = createdChannel?.channel?.better_channel_member.find(member => member.user_id !== req.userId)?.user_id;
+ const isBlocked = await UsersFunction.isUserBlocked(UserBlockedUser, req.userId, recipientId);
```

In the `UsersFunction` module, you would add a new function `isUserBlocked`:

```javascript
async function isUserBlocked(UserBlockedUser, userId, recipientId) {
  const result = await UserBlockedUser.findOne({
    where: { blockerId: userId, blockedId: recipientId }
  });
  return !!result;
}
```

This way, you're only fetching the necessary data and not the entire list of blocked users.


<!-- This is an auto-generated comment by OSS CodeRabbit -->